cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java

```diff
@@ -685,15 +685,17 @@ public boolean isOnline() {
       configureNetworkProxy();
       // we currently assume we have only a CLI process that runs shortly
       // therefore we run this check only once to save resources when this method is called many times
+      String url = "https://www.github.com";
       try {
         int timeout = 1000;
         //open a connection to github.com and try to retrieve data
         //getContent fails if there is no connection
-        URLConnection connection = new URL("https://www.github.com").openConnection();
+        URLConnection connection = new URL(url).openConnection();
         connection.setConnectTimeout(timeout);
         connection.getContent();
         this.online = Boolean.TRUE;
-      } catch (Exception ignored) {
+      } catch (Exception e) {
+        this.startContext.debug("Error when trying to connect to {}: {}", url, e);
```

Comment:
This will not log the exception properly. It will do `toString()` on the `Exception` what will be `«e.getClass()»: «e.getMessage()»`.
My suggestion was to do something like this:
```
if (debug().isEnabled()) {
  error(e, "Error when trying to connect to {}", url);
}
```

If you want to log with stacktrace on Debug, you can also do that like this:
```
debug().log(e, "Error when trying to connect to {}", url);
```